librz/core/vmlinux.c

```diff
@@ -0,0 +1,164 @@
+#include <rz_core.h>
+#include <rz_util/rz_str.h>
+#include <rz_util/rz_assert.h>
+#include "vmlinux.h"
+#include <stdio.h>
+
+static void add_config(RzVmlinuxConfigTable* config_tbl, char* config_name, char* config_value);
+
+RZ_API bool vmlinux_parse_apply_config_file(const char* config_filepath, RzVmlinuxConfigTable* config_tbl) {
+    rz_return_val_if_fail(config_filepath && config_tbl, false);
+
+    FILE* f = fopen(config_filepath, "r");
+    if (!f) {
+        return false;
+    }
+
+    size_t line_size = 256;
+    char* line = malloc(line_size);
+    if (!line) {
+        return false;
+    }
+
+    int err = 0;
+    while (true) {
+        ssize_t read_sz = getline(&line, &line_size, f); // reallocs if not enough
+        if (read_sz <= 0) {
+            err = 1;
+            goto out;
+        }
+
+        rz_str_trim(line);
+
+        if (!vmlinux_parse_apply_config_string(line, config_tbl)) {
+            RZ_LOG_WARN("Skipping line '%s'", line);
+        }
+    }
+out:
+    free(line);
+    fclose(f);
+    return !err;
+}
+
+RZ_API bool vmlinux_parse_apply_config_string(const char *config_str, RzVmlinuxConfigTable *config_tbl) {
+    rz_return_val_if_fail(config_str && config_tbl, false);
+
+    char* config_name;
+    char* config_value;
+
+    if (config_str[0] == '#' || !config_str[0]) {
+        return true;
+    } else if (rz_str_split_by_first_dupstr(config_str, "=", true, &config_name, &config_value)) {
+        add_config(config_tbl, config_name, config_value);
+        free(config_name);
+        free(config_value);
+        return true;
+    }
+    return false;
+}
+
+/**
+ * Numeric 90+
+*/
+RZ_API bool vmlinux_parse_version(const char* version_string, unsigned long version[3]) {
+    RzList* version_list = rz_str_split_duplist_n(version_string, ".", 3, true);
+    RzListIter* it;
+    char* str;
+
+    for (size_t i = 0; i < 3; ++i) version[i] = 0;
+
+    size_t v = 0;
+    rz_list_foreach(version_list, it, str) {
```

Comment:
Can you use `RzRegex` instead? In fact, it might be possible to use it for config file parsing as well, should simplify the code significantly

Reply:
fixed, thanks.